File: baekjoon/2606_바이러스/바이러스.py
# 실버 3 : https://www.acmicpc.net/problem/2606

from collections import deque

computers = int(input())
count = int(input())
graph = [[] for _ in range(computers + 1)]

for _ in range(count):
  x, y = map(int, input().split())
  graph[x].append(y)
  graph[y].append(x)

visited = [False]*(computers + 1)

result = 0

q = deque()
q.append(1)
visited[1] = True

while q:
  x= q.popleft()
  result += 1
  for i in graph[x]:
    if not visited[i]:
      q.append(i)
      visited[i] = True
print(result-1)

File: baekjoon/2606_바이러스/바이러스.js
// 실버 3 : https://www.acmicpc.net/problem/2606

const fs = require("fs");
const file = process.platform === "linux" ? "/dev/stdin" : "./text.txt";
const input = fs.readFileSync(file).toString().trim().split("\n");

const computers = Number(input[0]);
const count = Number(input[1]);

let graph = Array.from({ length: computers + 1 }, () => []);
for (let i = 2; i < input.length; i++) {
  let [x, y] = input[i].split(" ").map(Number);
  graph[x].push(y);
  graph[y].push(x);
}

let visited = Array(computers + 1).fill(false);
let q = [];
let result = 0;
q.push(1);
visited[1] = true;

while (q.length) {
  const x = q.shift();
  result += 1;
  for (const i of graph[x]) {
    if (!visited[i]) {
      q.push(i);
      visited[i] = true;
    }
  }
}
console.log(result - 1);
